src/commonMain/kotlin/fr/acinq/lightning/blockchain/electrum/ElectrumMiniWallet.kt

```diff
@@ -75,6 +76,19 @@ data class WalletState(val addresses: Map<String, List<Utxo>>) {
 
     companion object {
         val empty: WalletState = WalletState(emptyMap())
+
+        data class AddressState(val meta: AddressMeta, val alreadyUsed: Boolean, val utxos: List<Utxo>)
+
+        sealed class AddressMeta {
+            data object Single : AddressMeta()
+            data class Derived(val index: Int) : AddressMeta()
+        }
+
+        val AddressMeta.indexOrNull: Int?
```

Comment:
nit:  `indexOrNull`  could be a property of `AddressMeta` (so its type could be `Int` and not `Int?` in `Derived`)

Reply:
It would still have to be nullable though, because `Single` doesn't have one.